crates/cext/src/circuit.rs

```diff
@@ -1043,13 +1055,7 @@ pub unsafe extern "C" fn qk_circuit_delay(
     // SAFETY: Per documentation, the pointer is non-null and aligned.
     let circuit = unsafe { mut_ptr_as_ref(circuit) };
 
-    let delay_unit_variant = match unit {
-        QkDelayUnit::S => DelayUnit::S,
-        QkDelayUnit::MS => DelayUnit::MS,
-        QkDelayUnit::US => DelayUnit::US,
-        QkDelayUnit::NS => DelayUnit::NS,
-        QkDelayUnit::PS => DelayUnit::PS,
-    };
+    let delay_unit_variant = unit.into();
```

Comment:
Given the release deadline on this I'm fine with leaving this as is, but it is an unrelated change and shouldn't be in this PR.